test/common/event/scaled_range_timer_manager_impl_test.cc

```diff
@@ -316,8 +316,54 @@ TEST_F(ScaledRangeTimerManagerTest, SingleTimerSameMinMax) {
   EXPECT_FALSE(timer->enabled());
 }
 
+TEST_F(ScaledRangeTimerManagerTest, MaxOnlyScaleFactorTimer) {
+  ScaledRangeTimerManagerImpl manager(dispatcher_);
+
+  // With the default scale factor of 1, the timer should fire at its maximum timeout.
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(ScaleFactor(0.5), callback.AsStdFunction());
+  EXPECT_CALL(callback, Call());
+
+  timer->enableTimer(std::chrono::seconds(10));
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
```

Comment:
At the very least add between these 2 advances.
EXPECT_TRUE(timer->enabled());

It's unclear from the test what the registered min and max values are.  If the scale factor you passed in were interpreted as a millisecond number this test would still pass.